debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresDefaultValueConverterIT.java

```diff
@@ -121,6 +122,44 @@ public void testShouldHandleDefaultValueFunctionsWithSchemaPrefixes() throws Exc
         assertThat(defaultValue).isEqualTo("00000000-0000-0000-0000-000000000000");
     }
 
+    @Test
+    @FixFor("DBZ-7562")
+    public void shouldTruncateDefaultValuePrecisionToMatchColumnMaxPrecision() throws Exception {
+        final String ddl = "DROP SCHEMA IF EXISTS s1 CASCADE;"
+                + "CREATE SCHEMA s1;"
+                + "CREATE TABLE s1.dbz7562 (pk SERIAL, cost numeric(12,0) not null default 0.0, PRIMARY KEY(pk));"
+                + "INSERT INTO s1.dbz7562 (cost) values (2.25);"
+                + "INSERT INTO s1.dbz7562 (cost) values (3);"
+                + "INSERT INTO s1.dbz7562 (cost) values (0);";
+
+        // Test Snapshot
+        TestHelper.execute(ddl);
+
+        Configuration config = TestHelper.defaultConfig().build();
+        start(PostgresConnector.class, config);
+
+        waitForStreamingRunning("postgres", TestHelper.TEST_SERVER);
+
+        // Test Streaming
+        TestHelper.execute("INSERT INTO s1.dbz7562 (cost) values (5.5);");
+        TestHelper.execute("INSERT INTO s1.dbz7562 (cost) values (4);");
+        TestHelper.execute("INSERT INTO s1.dbz7562 (cost) values (0);");
+
+        final int expectedRecords = 6;
+        final SourceRecords records = consumeRecordsByTopic(expectedRecords);
+        final List<SourceRecord> recordsForTopic = records.recordsForTopic(topicName("s1.dbz7562"));
+        assertThat(recordsForTopic).hasSize(expectedRecords);
+
+        for (SourceRecord record : recordsForTopic) {
+            final Struct after = ((Struct) record.value()).getStruct(Envelope.FieldName.AFTER);
+            final Schema schema = after.schema().field("cost").schema();
+            assertThat(schema.parameters()).containsEntry("connect.decimal.precision", "12");
+            assertThat(schema.parameters()).containsEntry("scale", "0");
+            assertThat(schema.defaultValue().getClass()).isEqualTo(BigDecimal.class);
+            assertThat(schema.defaultValue()).isEqualTo(new BigDecimal(0).setScale(0));
```

Comment:
Until here you check if the schema is correct and has the expected types.
Shouldn't the streamed values maybe checked too? Especially in the case of BigDecimal and values like `0` and `0.0` etc. For checking the default value wouldn't it make sense to add another `INSERT` that add a record with the default value and see if it is streamed as `0` or `0.0`?

Reply:
But we don't trigger a default value entry, do we?

Reply:
@rk3rn3r no we don't but that isn't the reported problem.  The problem is with how we serialize the field's default value into `Field#defaultValue`, not into the field's value.

Reply:
Yes, but this test is the other side of the same thing, no? I don't know, I feel like it should then be tested there too (with a link to this DBZ issue) or testing the values here too. But I might be to strict on this. dunno.

Reply:
@jpechane wdyt?

Reply:
I just checked `RecordsStreamProducerIT` method `shouldReceiveChangesForTypeConstraints` where we explicitly set type as `numeric(6,1)` and insert a value of `2123.41` and we verify that the value is truncated and rounded down.  I can certainly add that here if we absolutely think its necessary but it seems redundant.